File: java/rna-transcription/src/main/java/RnaTranscription.java
import java.util.Map;
import java.util.HashMap;

class RnaTranscription {

    String transcribe(String dnaStrand) {
        String rnaTranscription = "";
        HashMap<Character, Character> nucleotides = new HashMap<>() {
            {
            put('G','C');
            put('C','G');
            put('T','A');
            put('A','U');
            }
        };
        for (int i = 0; i < dnaStrand.length(); i++) {
            rnaTranscription += nucleotides.get(dnaStrand.charAt(i));
        }
        return rnaTranscription;
    }
}

File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    def nucleotides(base):
        return {
            "G": "C",
            "C": "G",
            "T": "A",
            "A": "U"
        }.get(base)
    return ''.join( [nucleotides(base) for base in dna_strand ] )